secure/json-rpc/src/lib.rs

```diff
@@ -0,0 +1,595 @@
+    fn submit_transaction(&self, signed_transaction: SignedTransaction) -> Result<Response, Error> {
+        let method = "submit".to_string();
+        let params = vec![Value::String(hex::encode(lcs::to_bytes(
+            &signed_transaction,
+        )?))];
+        Ok(self.execute_request(method, params))
+    }
+
+    // Sends a get_account_state_with_proof() request to the JSON RPC server for the specified
+    // account, version height and ledger_version height (for the proof).
+    fn get_account_state_with_proof(
+        &self,
+        account: AccountAddress,
+        version: u64,
+        ledger_version: u64,
+    ) -> Result<Response, Error> {
+        let method = "get_account_state_with_proof".to_string();
+        let params = vec![
+            Value::String(account.to_string()),
+            Value::Number(version.into()),
+            Value::Number(ledger_version.into()),
+        ];
+        Ok(self.execute_request(method, params))
+    }
+}
+
+/// Below is a sample response from a successful submit() JSON RPC call:
+/// "{
+///   "id": 0,
+///   "jsonrpc": "2.0",
+///   "result": null
+/// }"
+///
+///
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct SubmitTransactionResponse {
+    id: u64,
+    jsonrpc: String,
+    result: Option<Value>,
+}
+
+/// Below is a sample response from a successful get_account_state_with_proof_call() JSON RPC call.
+/// "{
+///   "id": 0,
+///   "jsonrpc": "2.0",
+///   "result": {
+///     "blob": "0100...",
+///     "proof": {
+///       "ledger_info_to_transaction_info_proof": "00..",
+///       "transaction_info": "200000000000000000...<truncated>...ffffffffffffffff",
+///       "transaction_info_to_account_proof": "0000..."
+///     },
+///     "version": 0
+///   }
+/// }"
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct AccountStateWithProofResponse {
+    id: u64,
+    jsonrpc: String,
+    result: AccountStateResponse,
+}
+
+/// In practice this represents an AccountStateWithProof, however, we only decode the relevant
+/// fields here.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct AccountStateResponse {
+    version: u64,
+    blob: Option<Bytes>,
+}
+
+/// Below is a sample response from a failed JSON RPC call:
+/// "{
+///   "error": {
+///     "code": -32000,
+///     "data": null,
+///     "message": "Server error: send failed because receiver is gone"
+///   },
+///   "id": 0,
+///   "jsonrpc: "2.0"
+/// }"
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct JSONRpcFailureResponse {
+    id: u64,
+    jsonrpc: String,
+    error: JsonRpcError,
+}
+
+/// A custom error returned by the JSON RPC server for API calls that fail internally (e.g.,
+/// an internal error during execution on the server side).
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct JsonRpcError {
+    code: i16,
+    message: String,
+    data: Option<Value>,
+}
+
+/// Represents a vector of bytes using hex encoding.
```

Comment:
should we consider making this an LCS data structure Bytes<T> where T is Serialize / Deserialize? I am not sure if that would save you lines of code, but this is effectively an Hex(LCS) wrapper around types T.

Reply:
We could do that in the future, but seeing as though it won't save space, let's just keep it simple for now. Can always revisit it!